fenics_ice/inout.py

```diff
@@ -306,37 +306,85 @@ def field_from_vel_file(infile, field_name):
 
 def read_vel_obs(params, model=None):
     """
-    Read velocity observations & uncertainty from HDF5 file
-
-    For now, expects an HDF5 file
+    Reads velocity observations & uncertainty from a HDF5 file
+    containing:
+    - point cloud velocities (u_obs, v_obs) for inversion only
+    - gridded data set of composite ice velocities & uncertainties
+    use for alpha initialisation and boundary conditions
+    All variables in the HDF5 file should be with the form
+    (values, )
+
+    Generates self.u_obs, self.v_obs, self.u_std, self.v_std,
+    self.uv_obs_pts, self.mask_vel
+    self.u_comp, self.v_comp, self.u_comp_std, self.v_comp_std,
+    self.uv_comp_pts
     """
     infile = Path(params.io.input_dir) / params.obs.vel_file
     assert infile.exists(), f"Couldn't find velocity observations file: {infile}"
 
     infile = h5py.File(infile, 'r')
+    keys = list(infile.keys())
+
+    assert 'u_obs' in keys, \
+        f"Your vel file has not been configure correctly " \
+        f"make sure you have velocity data with the correct var names " \
+        f"x, y, u_obs, v_obs, u_obs_std, v_obs_std "
 
-    x_obs = field_from_vel_file(infile, 'x')
-    y_obs = field_from_vel_file(infile, 'y')
-    u_obs = field_from_vel_file(infile, 'u_obs')
-    v_obs = field_from_vel_file(infile, 'v_obs')
-    u_std = field_from_vel_file(infile, 'u_std')
-    v_std = field_from_vel_file(infile, 'v_std')
+    # Read composite mean of velocity data as default
     mask_vel = field_from_vel_file(infile, 'mask_vel')
+    x_comp = field_from_vel_file(infile, 'x')
+    y_comp = field_from_vel_file(infile, 'y')
+    u_comp = field_from_vel_file(infile, 'u_obs')
+    v_comp = field_from_vel_file(infile, 'v_obs')
+    u_comp_std = field_from_vel_file(infile, 'u_std')
+    v_comp_std = field_from_vel_file(infile, 'v_std')
```

Comment:
The variable names should probably match the keys here

Reply:
yeah the part of changing the keys inside the vel file is an easy fix but It would require that I also change the tests input data, the file `ismipc_U_obs.h5` used in here: https://github.com/EdiGlacUQ/fenics_ice/blob/main/tests/test_model.py#L105

has keys _obs ... (which is basically a composite mean from MEaSUREs) that is why I left _obs as a key for composite so no test would be broken. 

I can try to make a new file `ismipc_U_obs.h5` or changing the names of the existing one that I guess is also easy to do

Reply:
Actually @jrmaddison I think this is a bit more complicated than that .. I see now that other test: ismipc, ice-stream, etc also have their own velocity file with the key _obs defined. I would I think have to change a lot of independent files used by Joe to generate the different velocity files ... changing this var name inside the files could definitely break the tests

Reply:
never mind previous comments ... no need to change the vel file keys 😝 just need to change the variable name in the function...